File: src/main/java/org/labun/jooq/generator/util/TableNameConverter.java
package org.labun.jooq.generator.util;

import org.apache.commons.lang3.StringUtils;
import org.jooq.util.Definition;

import java.util.Arrays;
import java.util.stream.Collectors;

/**
 * @author Konstantin Labun
 */
public class TableNameConverter {

    public String convert(Definition element) {
        String name = element.getName();
        String[] words = name.split("_");

        return Arrays.stream(words)
                .map(String::toLowerCase)
                .map(StringUtils::capitalize)
                .collect(Collectors.joining(""));
    }
}

File: src/main/kotlin/org/labun/jooq/generator/util/TableNameConverter.kt
package org.labun.jooq.generator.util

import org.apache.commons.lang3.StringUtils
import org.jooq.util.Definition
import java.util.*
import java.util.stream.Collectors

/**
 * @author Konstantin Labun
 */
class TableNameConverter {

  fun convert(element: Definition): String {
    val name = element.name
    val words = name.split("_".toRegex()).dropLastWhile({ it.isEmpty() }).toTypedArray()

    return Arrays.stream<String>(words)
        .map { it.toLowerCase() }
        .map { StringUtils.capitalize(it) }
        .collect(Collectors.joining(""))
  }
}
